modules/library/http/src/main/java/org/geotools/http/SimpleHttpClient.java

```diff
@@ -120,13 +120,19 @@ public HTTPResponse post(
 
     private URLConnection openConnection(URL finalURL, Map<String, String> headers)
             throws IOException {
+        Map<String, Object> extraParams = getExtraParams();
+        if (extraParams != null) {
```

Comment:
See [do not return null](https://docs.geotools.org/latest/developer/conventions/code/null.html).

If you like we could update this convention to make use of Optional, as an explicit alternative to returning null.

Reply:
Done, changing default to Collections.emptyMap()